presto-hive/src/main/java/com/facebook/presto/hive/HiveClientConfig.java

```diff
@@ -1190,4 +1192,17 @@ public HiveClientConfig setS3SelectPushdownMaxConnections(int s3SelectPushdownMa
         this.s3SelectPushdownMaxConnections = s3SelectPushdownMaxConnections;
         return this;
     }
+
+    @Config("hive.temporary-staging-directory-enabled")
```

Comment:
How about we rename this to `hive.use-temporary-staging-directory-for-writes`?

Reply:
> But I don't feel we have to emphasize `for-write`? 

i thought the same

> What about `hive.use-temporary-staging-directory` ?

would do
however, i think the original `hive.temporary-staging-directory-enabled` is slightly better. Consider the case when one uses s3 where staging cannot (currently) be used.
When users sets `hive.use-temporary-staging-directory = true`, they could expect that staging is used (or error is reported). However, none of these will happen.
At the same time, I don't want to report an error when the option is used with s3 -- a single cluster may be talking to s3 and hdfs and user may want to set the option to change behavior on hdfs.

So... to me this should be either 
- `hive.temporary-staging-directory-enabled` (original) or
- `hive.hdfs.use-temporary-staging-directory` 

i prefer the original.

Reply:
Using `hdfs` could make someone to think that this only affects actual HDFS, and not any HDFS-compatible file system.

I would keep the original. @nezihyigitbasi What do you think?

Reply:
I feel like "using" a staging directory sounds better than "enabling" it (`hive.use-temporary-staging-directory`). But, if you all want to go with the original name I am fine with it.